src/osd/osd_types.cc

```diff
@@ -5826,9 +5827,10 @@ void SnapSet::encode(ceph::buffer::list& bl) const
 
 void SnapSet::decode(ceph::buffer::list::const_iterator& bl)
 {
-  DECODE_START_LEGACY_COMPAT_LEN(3, 2, 2, bl);
+  DECODE_START_LEGACY_COMPAT_LEN(4, 2, 2, bl);
   decode(seq, bl);
   bl += 1u;  // skip legacy head_exists (always true)
+  std::vector<snapid_t> snaps;  // removed in 4
   decode(snaps, bl);
```

Comment:
it should be checked for struct_v >=4 and skip the snaps, it will just leave a hole if not and we already bumped the version

Reply:
after talking to @Matan-B , i'm taking my words back, that change in the middle of the encode\decode, it will cause to backward compatibility issues. I'm with Matan solution for the dummy vector. LGTM